File: apps/mobile/src/screens/ProfileScreen.js
import { ScrollView, Text, View } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import MaterialIcons from '@expo/vector-icons/MaterialIcons';
import MaterialCommunityIcons from '@expo/vector-icons/MaterialCommunityIcons';
import { ChibiButton, ChibiSurface } from '../components/Chibi';
import { AvatarCanvas, selectionFromProfile } from '../components/Avatar';
import tokens from '../theme/tokens';
import { useAppStore } from '../state/useAppStore';

const GOAL_LABELS = {
  build_muscle: 'Build Muscle',
  lose_weight: 'Lose Weight',
  improve_health: 'Improve Health',
};

// No design/ mockup exists for this screen (only Personal.html, which is the
// onboarding "Your Stats" step already used by OnboardingStatsScreen) — this
// is an original layout built from the same tokens/components as the rest
// of the app rather than a ported mockup.
function StatTile({ icon, iconFamily = 'material', label, value, className = '' }) {
  const Icon = iconFamily === 'community' ? MaterialCommunityIcons : MaterialIcons;
  return (
    <ChibiSurface className={`flex-1 p-3 items-center ${className}`}>
      <Icon name={icon} size={20} color={tokens.colors.primary} />
      <Text className="font-headline text-lg text-on-background mt-1" numberOfLines={1}>
        {value}
      </Text>
      <Text className="font-label text-[10px] text-on-surface-variant uppercase text-center">{label}</Text>
    </ChibiSurface>
  );
}

export default function ProfileScreen() {
  const profile = useAppStore((s) => s.profile);
  const logout = useAppStore((s) => s.logout);
  const avatarSelection = selectionFromProfile(profile);
  const xpForNext = (profile?.level ?? 1) * 100;

  return (
    <SafeAreaView className="flex-1 bg-background">
      <ScrollView contentContainerClassName="p-margin-mobile gap-md" showsVerticalScrollIndicator={false}>
        <View className="items-center mb-1">
          <View className="flex-row items-center gap-2">
            <Text className="text-primary-container text-xl">✦</Text>
            <Text className="font-headline text-2xl text-on-background uppercase">Profile</Text>
            <Text className="text-primary-container text-xl">✦</Text>
          </View>
        </View>

        {/* Hero card — avatar + identity + XP */}
        <ChibiSurface className="items-center p-4">
          <View className="w-32 h-52 mb-2">
            <AvatarCanvas selection={avatarSelection} className="w-full h-full" />
          </View>
          <Text className="text-xl font-headline text-on-background" numberOfLines={1}>
            {profile?.displayName ?? 'Hero'}
          </Text>
          <View className="bg-secondary-container rounded-full px-3 py-1 border-[3px] border-ink mt-1 mb-3">
            <Text className="font-label text-xs text-on-secondary-container">LEVEL {profile?.level ?? 1}</Text>
          </View>
          <View className="w-full">
            <View className="flex-row justify-between mb-1">
              <Text className="font-label text-[11px] text-on-surface-variant uppercase">XP Progress</Text>
              <Text className="font-label text-[11px] text-primary">
                {profile?.xp ?? 0} / {xpForNext}
              </Text>
            </View>
            <View className="h-3 bg-surface-container-high rounded-full border-[2px] border-ink overflow-hidden">
              <View
                className="h-full bg-primary-container"
                style={{ width: `${Math.min(100, ((profile?.xp ?? 0) / xpForNext) * 100)}%` }}
              />
            </View>
          </View>
        </ChibiSurface>

        {/* Stat tiles */}
        <View className="flex-row gap-3">
          <StatTile icon="stars" label="Total XP" value={profile?.totalXp ?? 0} />
          <StatTile icon="monetization-on" label="Coins" value={profile?.coins ?? 0} />
          <StatTile icon="diamond-stone" iconFamily="community" label="Gems" value={profile?.gems ?? 0} />
        </View>

        {/* Goal + body stats */}
        <ChibiSurface className="p-4">
          <View className="flex-row items-center gap-2 mb-3">
            <MaterialIcons name="flag" size={18} color={tokens.colors['primary-container']} />
            <Text className="font-label text-[12px] uppercase text-on-surface-variant">Current Goal</Text>
          </View>
          <Text className="font-headline text-lg text-on-background uppercase mb-4">
            {GOAL_LABELS[profile?.goal] ?? '—'}
          </Text>

          <View className="flex-row gap-3">
            <View className="flex-1">
              <View className="flex-row items-center gap-1 mb-1">
                <MaterialIcons name="accessibility-new" size={14} color={tokens.colors['primary-container']} />
                <Text className="font-label text-[10px] uppercase text-on-surface-variant">Height</Text>
              </View>
              <Text className="font-headline text-on-background">
                {profile?.heightCm ? `${profile.heightCm} cm` : '—'}
              </Text>
            </View>
            <View className="flex-1">
              <View className="flex-row items-center gap-1 mb-1">
                <MaterialIcons name="monitor-weight" size={14} color={tokens.colors['primary-container']} />
                <Text className="font-label text-[10px] uppercase text-on-surface-variant">Weight</Text>
              </View>
              <Text className="font-headline text-on-background">
                {profile?.weightKg ? `${profile.weightKg} kg` : '—'}
              </Text>
            </View>
          </View>
        </ChibiSurface>

        <ChibiButton className="py-4 bg-error-container" onPress={logout}>
          <Text className="font-headline uppercase text-on-error-container">Log out</Text>
        </ChibiButton>
      </ScrollView>
    </SafeAreaView>
  );
}

File: fitQuest/fitQuest/Screens/ProfileScreen.swift
//
//  ProfileScreen.swift
//  fitQuest
//
//  Ported from screens/ProfileScreen.js (redesigned version: hero avatar +
//  XP card, stat tiles row, goal/height/weight card). No design/ mockup
//  exists for this screen — it's an original layout using the same tokens
//  as the rest of the app.
//

import SwiftUI

private let goalLabels: [String: String] = [
    "build_muscle": "Build Muscle",
    "lose_weight": "Lose Weight",
    "improve_health": "Improve Health",
]

struct ProfileScreen: View {
    @EnvironmentObject private var appState: AppState

    private var profile: Profile? { appState.profile }
    private var xpForNext: Int { (profile?.level ?? 1) * 100 }
    private var xpProgress: Double {
        guard xpForNext > 0 else { return 0 }
        return min(1, Double(profile?.xp ?? 0) / Double(xpForNext))
    }
    private var goalLabel: String {
        profile?.goal.flatMap { goalLabels[$0] } ?? "—"
    }

    var body: some View {
        ScrollView {
            VStack(spacing: Tokens.Spacing.md) {
                HStack(spacing: Tokens.Spacing.base) {
                    Text("✦").foregroundStyle(Tokens.Colors.primaryContainer)
                    Text("PROFILE").font(.title2.bold())
                    Text("✦").foregroundStyle(Tokens.Colors.primaryContainer)
                }
                .padding(.top, Tokens.Spacing.sm)

                heroCard
                statTiles
                goalAndStatsCard

                Button {
                    appState.logout()
                } label: {
                    Text("Log out")
                        .font(.headline)
                        .textCase(.uppercase)
                        .foregroundStyle(Tokens.Colors.onErrorContainer)
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, Tokens.Spacing.sm)
                }
                .buttonStyle(.chibi(fill: Tokens.Colors.errorContainer))
            }
            .padding(Tokens.Spacing.md)
        }
        .background(Tokens.Colors.background)
    }

    private var heroCard: some View {
        ChibiSurface {
            VStack(spacing: Tokens.Spacing.base) {
                AvatarCanvasView(selection: avatarSelection(from: profile))
                    .frame(width: 128, height: 208)

                Text(profile?.displayName ?? "Hero")
                    .font(.title3.bold())
                    .lineLimit(1)

                Text("LEVEL \(profile?.level ?? 1)")
                    .font(.system(size: 11, weight: .bold))
                    .padding(.horizontal, Tokens.Spacing.sm)
                    .padding(.vertical, 4)
                    .background(Tokens.Colors.secondaryContainer)
                    .foregroundStyle(Tokens.Colors.onSecondaryContainer)
                    .clipShape(Capsule())
                    .overlay(Capsule().stroke(Tokens.Colors.ink, lineWidth: 3))

                VStack(alignment: .leading, spacing: 4) {
                    HStack {
                        Text("XP PROGRESS").font(.system(size: 11, weight: .bold)).foregroundStyle(Tokens.Colors.onSurfaceVariant)
                        Spacer()
                        Text("\(profile?.xp ?? 0) / \(xpForNext)")
                            .font(.system(size: 11, weight: .bold))
                            .foregroundStyle(Tokens.Colors.primary)
                    }
                    GeometryReader { proxy in
                        ZStack(alignment: .leading) {
                            Capsule().fill(Tokens.Colors.surfaceContainerHigh)
                            Capsule().fill(Tokens.Colors.primaryContainer)
                                .frame(width: proxy.size.width * xpProgress)
                        }
                    }
                    .frame(height: 12)
                    .overlay(Capsule().stroke(Tokens.Colors.ink, lineWidth: 2))
                }
                .frame(maxWidth: .infinity)
            }
            .padding(Tokens.Spacing.md)
        }
    }

    private var statTiles: some View {
        HStack(spacing: Tokens.Spacing.sm) {
            statTile(icon: "star.fill", label: "Total XP", value: "\(profile?.totalXp ?? 0)")
            statTile(icon: "dollarsign.circle.fill", label: "Coins", value: "\(profile?.coins ?? 0)")
            statTile(icon: "diamond.fill", label: "Gems", value: "\(profile?.gems ?? 0)")
        }
    }

    private func statTile(icon: String, label: String, value: String) -> some View {
        ChibiSurface {
            VStack(spacing: 4) {
                Image(systemName: icon).foregroundStyle(Tokens.Colors.primary)
                Text(value).font(.headline).lineLimit(1)
                Text(label.uppercased())
                    .font(.system(size: 9, weight: .bold))
                    .foregroundStyle(Tokens.Colors.onSurfaceVariant)
                    .multilineTextAlignment(.center)
            }
            .frame(maxWidth: .infinity)
            .padding(Tokens.Spacing.sm)
        }
    }

    private var goalAndStatsCard: some View {
        ChibiSurface {
            VStack(alignment: .leading, spacing: Tokens.Spacing.base) {
                HStack(spacing: 4) {
                    Image(systemName: "flag.fill").foregroundStyle(Tokens.Colors.primaryContainer)
                    Text("CURRENT GOAL")
                        .font(.system(size: 12, weight: .bold))
                        .foregroundStyle(Tokens.Colors.onSurfaceVariant)
                }
                Text(goalLabel.uppercased())
                    .font(.title3.bold())

                HStack(spacing: Tokens.Spacing.sm) {
                    bodyStat(icon: "figure.arms.open", label: "Height", value: profile?.heightCm.map { "\(Int($0)) cm" } ?? "—")
                    bodyStat(icon: "scalemass.fill", label: "Weight", value: profile?.weightKg.map { "\(Int($0)) kg" } ?? "—")
                }
            }
            .padding(Tokens.Spacing.md)
        }
    }

    private func bodyStat(icon: String, label: String, value: String) -> some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: 4) {
                Image(systemName: icon).font(.caption).foregroundStyle(Tokens.Colors.primaryContainer)
                Text(label.uppercased()).font(.system(size: 10, weight: .bold)).foregroundStyle(Tokens.Colors.onSurfaceVariant)
            }
            Text(value).font(.headline)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
    }
}

#Preview {
    ProfileScreen()
        .environmentObject(AppState())
}
